### PR TITLE
Fix bug for Qiskit circuits with idle qubits

### LINKED ISSUE #1255: Fix idle qubit register indexing
Pre-Report Checklist
--------------------

- [x] I am running the latest version of mitiq
- [x] I checked to make sure that this bug has not already been reported

Issue Description
-----------------

Qiskit circuits built on registers like `qiskit.QuantumRegister(1) for _ in range(4)` insert (and remove) idle identities poorly, in particular when combined with DDD. Qubit index returns `0` for registers of the given type (and so other qubits are altered).

How to Reproduce
----------------

### Code Snippet

```python
import qiskit
from mitiq.ddd.insertion import insert_ddd_sequences
from mitiq.ddd.rules import xx, xyxy

qreg = [qiskit.QuantumRegister(1) for _ in range(4)]
circuit_qiskit_one = qiskit.QuantumCircuit(*qreg)
for q in qreg:
    circuit_qiskit_one.x(q)
    circuit_qiskit_one.x(3)

print(circuit_qiskit_one)
print(insert_ddd_sequences(circuit_qiskit_one, xyxy))
```
### Error Output
```
    ┌───┐                    
q0: ┤ X ├────────────────────
    ├───┤                    
q1: ┤ X ├────────────────────
    ├───┤                    
q2: ┤ X ├────────────────────
    ├───┤┌───┐┌───┐┌───┐┌───┐
q3: ┤ X ├┤ X ├┤ X ├┤ X ├┤ X ├
    └───┘└───┘└───┘└───┘└───┘
    ┌───┐┌───┐┌───┐┌───┐┌───┐┌───┐
q0: ┤ X ├┤ I ├┤ X ├┤ Y ├┤ X ├┤ Y ├
    ├───┤├───┤├───┤├───┤├───┤├───┤
q1: ┤ X ├┤ I ├┤ X ├┤ Y ├┤ X ├┤ Y ├
    ├───┤├───┤├───┤├───┤├───┤├───┤
q2: ┤ X ├┤ I ├┤ X ├┤ Y ├┤ X ├┤ Y ├
    ├───┤├───┤├───┤├───┤├───┤├───┤
q3: ┤ X ├┤ X ├┤ X ├┤ X ├┤ X ├┤ I ├
    └───┘└───┘└───┘└───┘└───┘└───┘

### PR DESCRIPTION
**Pull request by andreamari**
- Fixes #1255

The problem was caused by using integer indices for identifying idle qubits.
The problem is now fixed by using `qiskit.circuit.Qubit` objects instead of integer indeces.

The implemented solution doesn't use a Qiskit Transpiler pass as originally discussed, due to the 
complexity of defining a custom transpiler.   

### License

- [x ] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

### CODE REVIEW

File: mitiq/interface/mitiq_qiskit/conversions.py [hunk 1]
```diff
@@ -133,45 +133,42 @@ def _add_identity_to_idle(
...
 def _remove_identity_from_idle(
     circuit: qiskit.QuantumCircuit,
-    idle_indices: Set[int],
+    idle_qubits: Set[qiskit.circuit.Qubit],
 ) -> None:
-    """Removes identities from the circuit corresponding to the set of indices.
+    """Removes identities from the circuit corresponding to the input
+    idle qubits.
     Used in conjunction with _add_identity_to_idle to preserve idle qubits and
     indices in conversion.
```
Comment (natestemen) [inline mitiq/interface/mitiq_qiskit/conversions.py:158]:
```suggestion
    Used in conjunction with _add_identity_to_idle to preserve idle qubits in
    conversion.
```

That's not true anymore, right?

**Review by natestemen — APPROVED** [review]
Nice work Andrea! It'll be nice to have this issue closed.

File: mitiq/interface/conversions.py [hunk 2]
```diff
@@ -234,7 +234,6 @@ def new_scaling_function(
     ) -> QPROGRAM:
 
         # Pre atomic conversion
-        idle_indices = set()
         if "qiskit" in circuit.__module__:
```
[the remark this answers is not in the document]
Comment (andreamari) [reply mitiq/interface/conversions.py:237]: In principle, `scaled_circuit` should always have the same same type as `circuit`. 
But, just to be sure, I replaced `scaled_circuit` with `circuit`.

[the remark this answers is not in the document]
Comment (natestemen) [reply mitiq/interface/conversions.py:237]: Looks like the tests are failing after that change. I guess we should stick with `scaled_circuit`, although I don't fully understand why.

BTW the docs build failure is due to a broken link that should be resolved soon, once the new RTD latest build is published.

[the remark this answers is not in the document]
Comment (natestemen) [reply mitiq/interface/conversions.py:237]: Andrea and I discussed this live, and agree that this is safe to remove.

**Pull request merged**